From 40bfaf8acd30c5191b94f0c1e8e514dbbdbc3d9f Mon Sep 17 00:00:00 2001
From: Tom Gundersen <teg@jklm.no>
Date: Wed, 8 Jul 2015 13:52:48 +0200
Subject: [PATCH] resolved: fix marshalling of RRSIG records

The key tag is 16, not 8 bits.

(cherry picked from commit 0b1b17d3256c409254b114d63b4ed2308c10996d)
---
 src/resolve/resolved-dns-packet.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/resolve/resolved-dns-packet.c b/src/resolve/resolved-dns-packet.c
index 21756f566f..8a4d029780 100644
--- a/src/resolve/resolved-dns-packet.c
+++ b/src/resolve/resolved-dns-packet.c
@@ -691,7 +691,7 @@ int dns_packet_append_rr(DnsPacket *p, const DnsResourceRecord *rr, size_t *star
                 if (r < 0)
                         goto fail;
 
-                r = dns_packet_append_uint8(p, rr->rrsig.key_tag, NULL);
+                r = dns_packet_append_uint16(p, rr->rrsig.key_tag, NULL);
                 if (r < 0)
                         goto fail;
 
